### PR TITLE
Fix #9305 LaTeX: backslash in sphinxupquote error with Japanese

### LINKED ISSUE #9305: sphinxupquote and textbackslash: Improper discretionary list.
**Describe the bug**
Related to: https://github.com/python/python-docs-ja/issues/22

While trying to reproduce https://github.com/python/python-docs-ja/issues/22 I was able to produce:

```latex
%% Generated by Sphinx.
\def\sphinxdocclass{jsbook}
\documentclass[a4paper,10pt,dvipdfmx]{sphinxmanual}
\usepackage{sphinx}
\usepackage{hyperref}
\begin{document}
\sphinxupquote{\textbackslash{}x}
\end{document}
```

It is manually produced from `c-api.tex` from a cpython documentation japanese build, by manually bisecting the file to find a minimal reproducer.

**To Reproduce**

- Write the LaTeX example to a foo.tex file
- run `/usr/bin/platex -kanji=utf8 -recorder foo.tex`

**Environment info**
- OS: Debian 11
- Python version: 3.9.4
- Sphinx version: 3.2.1
- Extra tools: 
```
$ /usr/bin/platex --version
e-pTeX 3.14159265-p3.8.3-191112-2.6 (utf8.euc) (TeX Live 2020/Debian)
kpathsea version 6.3.2
ptexenc version 1.3.8
```
 
**Additional context**

I don't understand the error message, I don't know how to tell if it's a Sphinx issue (that should not have generated this) or a platex issue (that should understand this), sry.

### PR DESCRIPTION
**Pull request by jfbu**
### Relates
- Fix #9305

### CODE REVIEW

File: sphinx/texinputs/sphinxlatexliterals.sty [hunk 1]
```diff
@@ -765,7 +765,8 @@
       % break at . , ; ? ! /
       \sphinxbreaksviaactive
       % break also at \
-      \let\sphinx@textbackslash\textbackslash
+      \setbox8=\hbox{\textbackslash}%
+      \def\sphinx@textbackslash{\copy8}%
```
Comment (jfbu) [inline sphinx/texinputs/sphinxlatexliterals.sty:769]: I should perhaps use a proper scratch box but the contents of `\sphinxupquote` should be a text string hence it seems legit to use say `\box8` as here (as this is enclosed in braces).

The `\textbackslash` is a complex command which checks the current font encoding and triggers a fall-back font encoding (OMS, math character) for example if character not available. So the former method with `\let` did not have expected effect and the `\textbackslash` was trying to redefine itself in a  `\discretionary` context where only boxes are allowed.

**Comment by jfbu** [comment]
I used this test file from a former issue with backslash and line-wrapping in literal environments.

```rest
Welcome to FOO's documentation!
===============================

:file:`a\\b`

:file:`\\C\\a{}a{}a{}\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa\\aaa`

::

   \aaa

``\aaa``

:file:`\\C\\aaa`


::

   \a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\a\

AAA

.. parsed-literal::

   \\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\a\\

AAA

.. raw:: latex

         \par\hrule\par

:file:`a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\a\\\\aa`

```

I tested with `language='ja'` and it works.

File: sphinx/texinputs/sphinxlatexliterals.sty (hunk 1 above)
Comment (jfbu) [reply sphinx/texinputs/sphinxlatexliterals.sty:769]: (I mentioned OMS from my pure latex testing, but in Sphinx context this will be T1 encoding)

**Review by tk0miya — APPROVED** [review]

**Comment by jfbu** [comment]
Thanks for review, I merge now.

**Pull request merged**